src/src/PluginStructs/P145_data_struct.cpp

```diff
@@ -0,0 +1,655 @@
+/*!
+@brief  Initialize the pluging stuct/class 
+@return always true
+@note   This function is expected to be called as part of the plugin 
+        PLUGIN_INIT functionality
+*/
+/*****************************************************************************/
+bool P145_data_struct::plugin_init()
+{
+  resetOversampling();
+  last_ain = 0;                   // No measured analog input value yet
+  last_cal = millis();            // Initialise the calibration interval
+  return true;
+}
+
+/*****************************************************************************/
+/*!
+@brief  Handle the 10Hz processing for the plugin strct/class 
+@return Always true
+@note   This function is expected to be called as part of the plugin 
+        PLUGIN_TEN_PER_SECOND functionality
+*/
+/*****************************************************************************/
+bool P145_data_struct::plugin_ten_per_second()
+{
+    uint16_t currentValue;
+
+    // Measure the analog input value for oversampling
+    // Algorithm uses mean value with exeption of max and min values
+    // See declaraion of the global variables for their usage
+    // Skip measurement in analog input is used to calibrate WiFi
+    if (!WiFiEventData.wifiConnectInProgress)
+    {
+      currentValue = espeasy_analogRead(analogPin);
+      ovs_value += currentValue;
+      ovs_cnt++;
+
+      if (currentValue > ovs_max)
+      {
+        ovs_max = currentValue;
+      }
+      if (currentValue < ovs_min)
+      {
+        ovs_min = currentValue;
+      }
+    }
+    return(true);
+}
+
+/**************************************************************************/
+/*!
+@brief  Return the actual Calibration value. 
+@return Rzero associated with the current measurement
+@note   The value depends on the conversion algorithm associated with the 
+        sensor type.
+*/
+/**************************************************************************/
+float P145_data_struct::getCalibrationValue()
+{
+  return (rcal);
+}
+
+/**************************************************************************/
+/*!
+@brief  Return the Rzero value determined by autocalibration. 
+@return Rzero as calculated by the autocalibration algorithm
+@note   This is the internally corrected Rzero and will be unaltered when
+        Autocalibration is switched off.
+*/
+float P145_data_struct::getAutoCalibrationValue()
+{
+  return rzero;
+}
+
+/**************************************************************************/
+/*!
+@brief  Dump data from the internal plugin struct/class structure to the log
+@note   This function is only used for debugging
+*/
+/**************************************************************************/
+void P145_data_struct::dump()
+{
+  if (loglevelActiveFor(LOG_LEVEL_INFO))
+  {
+#ifdef P145_DEBUG
+    String log;                       // Helper string to build log text
+    log = F("MQ-xx: NAME ");
+    log += sensordef.name;
+    addLog(LOG_LEVEL_INFO, log);
```

Comment:
Why not like this:
```c++
    addLog(LOG_LEVEL_INFO, concat(F("MQ-xx: NAME "), sensordef.name));
```
This way there is no need for allocating a `log` variable, it makes the code more readable (in this use case) and the return value of `concat` will be moved using `std::move` instead of being copied.